src/components/data/DataTable.js

```diff
@@ -114,6 +114,16 @@ const DataTable = props => {
           { billingProject: googleProject, dataReference: snapshotName } :
           { filterTerms: activeTextFilter })
       }))
+    // find all the unique attribute names contained in the current page of results
+    const attrNamesFromResults = _.uniq(_.flatMap(_.keys, _.map('attributes', results)))
+    // add any attribute names from the current page of results to those found in metadata.
+    // This allows for the stale metadata (e.g. the metadata cache is out of date).
+    // For the time being, the uniquness check MUST be case-insensitive (e.g. { sensitivity: 'accent' })
+    // in order to prevent case-divergent columns from being displayed, as that would expose some other bugs
+    const newAttrsForThisType = concatenateAttributeNames(entityMetadata[entityType]?.attributeNames, attrNamesFromResults)
+    if (!_.isEqual(newAttrsForThisType, entityMetadata[entityType].attributeNames)) {
+      setEntityMetadata(_.set([entityType, 'attributeNames'], newAttrsForThisType))
+    }
```

Comment:
Consider

```suggestion
    // Find all the unique attribute names contained in the current page of results.
    const attrNamesFromResults = _.uniq(_.flatMap(_.keys, _.map('attributes', results)))
    // Add any attribute names from the current page of results to those found in metadata.
    // This allows for the stale metadata (e.g. the metadata cache is out of date).
    // For the time being, the uniqueness check MUST be case-insensitive (e.g. { sensitivity: 'accent' })
    // in order to prevent case-divergent columns from being displayed, as that would expose some other bugs
    const  attrNames = entityMetadata[entityType]?.attributeNames
    const newAttrsForThisType = concatenateAttributeNames(attrNames, attrNamesFromResults)
    if (!_.isEqual(newAttrsForThisType, attrNames)) {
      setEntityMetadata(_.set([entityType, 'attributeNames'], newAttrsForThisType))
    }
```

Reply:
GitHub didn't diff this the way I was hoping. The diff includes:
- correction of a typo
- capitalization of first letter of comment lines
- pulling out `attrNames` to be on the subsequent lines